scrapy/http/request/form.py

```diff
@@ -33,7 +33,14 @@ def __init__(self, *args, **kwargs):
                 self.headers.setdefault(b'Content-Type', b'application/x-www-form-urlencoded')
                 self._set_body(querystr)
             else:
-                self._set_url(self.url + ('&' if '?' in self.url else '?') + querystr)
+                url_split = urlsplit(self.url)
+                formdata_key_list = []
+                for k in querystr.split('&'):
+                    formdata_key_list.append(k.split('=')[0])
+                items = []
+                items += [(k, v) for k, v in parse_qsl(url_split.query) if k not in formdata_key_list]
```

Comment:
Why create an empty ``items`` here and then add to this list? Can this be reduced to one operation?

Reply:
Here ```items``` is a local variable, since this is the process of appending to a empty list. If we write simply ```items+=[........]``` it will be error as ```local variable referenced before assignment```

Reply:
:+1: